test/framework/src/main/java/org/elasticsearch/cluster/routing/TestShardRouting.java

```diff
@@ -217,20 +223,38 @@ public static UnassignedInfo buildUnassignedInfo(String message) {
             lastAllocatedNodeId = randomIdentifier();
         }
         int failedAllocations = reason == UnassignedInfo.Reason.ALLOCATION_FAILED ? 1 : 0;
+
+        long unassignedTimeMillis = randomLongBetween(
+            1672531200000L, // 01-01-2023
+            1704067200000L // 01-01-2024
+        );
```

Comment:
This could be any (nonnegative) `long` couldn't it?

Reply:
Another assertion that this might fail is inside of TimeValue constructor: https://github.com/elastic/elasticsearch/blob/997fbb372cdec82eb803d32de5147dbccec21e64/server/src/main/java/org/elasticsearch/rest/action/cat/RestShardsAction.java#L329

Reply:
I think that's not an assertion error, just an `IllegalArgumentException`, and it seems like a genuine bug too. It's entirely possible that the coordinating node's `System.currentTimeMillis()` is behind the master's clock which was the source of `shard.unassignedInfo().getUnassignedTimeInMillis()`.